src/Record/Record.Model/Immutable/Record.cs

```diff
@@ -118,6 +140,72 @@ public IGraph MetadataGraph()
         return tempGraph;
     }
 
+    private void AssertDescribesConstraint()
+    {
+        if (!AskIfDescribedObjectExistOnContentGraph())
+            throw new RecordException("All described objects on the metadata graph must exist as subjects on the content graph.");
+
+        if (AskIfContentSubjectIsUnreachableFromMetadata())
+            throw new RecordException("All nodes on the content graph must be reachable through the describes predicate on the metadata graph.");
+    }
+
+    private bool AskIfDescribedObjectExistOnContentGraph()
+    {
+        var parameterizedQuery = new SparqlParameterizedString(@"
+            ASK {
+                GRAPH ?metaGraph {
+                    ?meta a @Record; 
+                         @describes ?object;
+                         @hasContent ?content.}
+                { GRAPH ?content {?desc ?P ?O.} }
+                    UNION 
+                { GRAPH ?content {?S ?P ?desc} }
+            }");
+
+        parameterizedQuery.SetUri("Record", new Uri(Namespaces.Record.RecordType));
+        parameterizedQuery.SetUri("describes", new Uri(Namespaces.Record.Describes));
+        parameterizedQuery.SetUri("hasContent", new Uri(Namespaces.Record.HasContent));
+
+
+        var parser = new SparqlQueryParser();
+        var queryString = parameterizedQuery.ToString();
+        var query = parser.ParseFromString(queryString);
+
+        var queryResult = (SparqlResultSet)_queryProcessor.ProcessQuery(query);
+        return queryResult.Result;
+    }
+
+    private bool AskIfContentSubjectIsUnreachableFromMetadata()
+    {
+        var parameterizedQuery = new SparqlParameterizedString(@"
+            ASK {
+                GRAPH ?metaGraph {
+                    ?recId a @Record ;
+                    @hasContent ?content. 
+                }
+                { GRAPH ?content {?unreachable ?P ?O. } }
+                    UNION
+                { GRAPH ?content {?S ?P ?unreachable . } }
+                
+                FILTER NOT EXISTS {
+                    GRAPH ?metaGraph { ?recId @describes ?describedObject. }
+                    GRAPH ?content {?describedObject !@notConnected* ?unreachable . }
```

Comment:
I may be misunderstanding something here, but do you not need this going the other way as well? Incase `?unreachable` only exists as a subject.

I.e. that we need `?unreachable !@notConnected* ?describedObject` in addition to the statment above. 🤔